app/controllers/tasks_controller.rb

```diff
@@ -1,80 +1,133 @@
 class TasksController < ApplicationController
 
   def index
-    @all_tasks = Task.all   # NOTE TO SELF: "task" is actually a Task::ActiveRecord_Relation object, but responds to Array methods
-    completed_instance_var
+    @all_tasks = Task.all
+      # NOTE TO SELF: "@all_tasks" is actually a Task::ActiveRecord_Relation object, but responds to array methods
 
-    render :index   # for clarity, you don't actually need this in this case
+    render :index
+      # NOTE TO SELF: this is for clarity, you don't actually need this in this case
+  end
+
+  def completed_status
+    @task = Task.find(params[:id])
+    if @task[:completed_at].nil?
+      Task.completed(params[:id])
+    else
+      Task.not_completed(params[:id])
+    end
+
+    redirect_to "/"
   end
 
   def show
-    task_instance_vars
-    @visibility = "invisible"
+    task = Task.find(params[:id])
+
+    @task_id =              task[:id]
+    @task_name =            task[:name]
+    @task_description =     task[:description]
+    @task_status =          task[:completed_at].nil? ? "NEED TO PLAY" : "FINISHED"
```

Comment:
This is a tricky one - if these strings are used for display purposes, it's probably better to keep this logic in the view, rather than the controller.